### PR TITLE
Add Searching function for Data Sets

### LINKED ISSUE #2095: Search for a string in members of a PDS
**Is your feature or enhancement request related to a problem or limitation? Please describe**

Zowe CLI and Explorer users do not have an automated way to search for a string in members of a PDS. Currently, the only options are to download the entire PDS to a local hard drive and perform a search locally or go into ISPF and perform a SRCHFOR operation there. 

**Describe your enhancement idea**

It would be useful to have something similar to the `zowe jobs search job` command but for PDSs. Ideally, this functionality could eventually be used by Zowe Explorer, where users are more likely to be interacting directly with their data sets.

**Describe alternatives you've considered**

- Download and search locally (requires downloading an entire PDS, which may be undesirable in terms of network/disk usage and time)
- ISPF (fast, but requires the user to log in to a 3270 terminal)
- Submitting JCL to perform a search (fast, but requires updating JCL for each search)

**Provide any additional context**

### PR DESCRIPTION
**Pull request by awharn**
**What It Does**

Adds the ability to search data sets for a search term
Resolves #2095

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

**Comment by awharn** [comment]
I have opened the PR for reviews. Please do not merge until we have had a discussion regarding the CLI output. I plan to demo this functionality and discuss on Monday's stand-up meeting.

File: packages/cli/src/zosfiles/download/Download.options.ts [hunk 1]
```diff
@@ -16,7 +16,8 @@ import i18nTypings from "../-strings-/en";
 // Does not use the import in anticipation of some internationalization work to be done later.
 const strings = (require("../-strings-/en").default as typeof i18nTypings).DOWNLOAD.OPTIONS;
 
-const maxConcurrentRequestsMaxValue = 99999;
+/** @internal */
+export const maxConcurrentRequestsMaxValue = 99999;
```
Comment (awharn) [inline packages/cli/src/zosfiles/download/Download.options.ts:20]: Do we want to decrease this number for a future PR? If so, any suggestions on what the limit should be?

Comment (zFernand0) [reply packages/cli/src/zosfiles/download/Download.options.ts:20]: maybe 100 (if it's too low, then 1000) 😋

Comment (adam-wolfe) [reply packages/cli/src/zosfiles/download/Download.options.ts:20]: 100 seems reasonable. I tried 1000 and would not recommend it.

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 
image

**Review by traeok — APPROVED** [review]
LGTM, thanks for the useful enhancement @awharn!
I left a couple comments, but I didn't notice anything that needs changed.

File: packages/cli/src/zosfiles/search/ds/DataSets.definition.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    },
+    positionals: [
+        {
+            name: "pattern",
+            description: dataSetStrings.POSITIONALS.PATTERN,
+            type: "string",
+            required: true
+        },
+        {
+            name: "search-string",
```
Comment (t1m0thyj) [inline packages/cli/src/zosfiles/search/ds/DataSets.definition.ts:42]: Typically I think we use camel case (e.g., `searchString`) for positional names.

**Comment by awharn** [comment]
Well, I accidentally hit merge here instead of in another PR. Will address review comments in a new PR shortly. In the meantime, I  have stopped workflows to prevent publication.

**Pull request merged**